### PR TITLE
Add support for an ordered listing

### PR DESCRIPTION
**Pull request by thomasjungblut**
Hey,

apologies in advance for the random PR, I'm coming from the discussion at [gatsby-remark-table-of-contents-8](https://github.com/signalwerk/gatsby-remark-table-of-contents/issues/8), which is a pretty simple wrapper around the MDAST for GatsbyJS.

The goal is to pass a new "ordered" setting around and manifest it in the whole TOC tree.  
Happy to make any adjustments. 

Cheers,
Thomas

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 .DS_Store
+.history
```
Comment (wooorm) [inline .gitignore:2]:
```suggestion
```

Unrelated change

File: lib/contents.js [hunk 2]
```diff
@@ -5,8 +5,8 @@ var extend = require('extend')
 module.exports = contents
 
 // Transform a list of heading objects to a markdown list.
-function contents(map, tight, prefix) {
-  var table = {type: 'list', ordered: false, spread: false, children: []}
+function contents(map, tight, prefix, ordered = false) {
+  var table = {type: 'list', ordered: ordered, spread: false, children: []}
```
Comment (wooorm) [inline lib/contents.js:9]:
```suggestion
function contents(map, tight, prefix, ordered) {
  var table = {type: 'list', ordered: ordered, spread: false, children: []}
```

File: lib/index.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ function toc(node, options) {
   var result = search(node, heading, settings)
 
   result.map = result.map.length
-    ? contents(result.map, settings.tight, settings.prefix)
+    ? contents(result.map, settings.tight, settings.prefix, settings.ordered)
```
Comment (wooorm) [inline lib/index.js:16]:
```suggestion
    ? contents(result.map, settings.tight, settings.prefix, settings.ordered || false)
```

File: test/fixtures/headings-ordered/config.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "ordered": true
+}
```
Comment (wooorm) [inline test/fixtures/headings-ordered/config.json:3]: missing an EOF EOL

**Review by wooorm — COMMENTED** [review]
great idea!

**Comment by thomasjungblut** [comment]
Thanks for the review @wooorm, everything applied.

**Review by BarryThePenguin — APPROVED** [review]
👍 looks great! 🎉

**Comment by thomasjungblut** [comment]
thanks!, let's merge? 🚀

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Thanks @thomasjungblut, released!
